implementations/rust/vault/src/lib.rs

```diff
@@ -102,33 +98,31 @@ pub trait Vault: Zeroize {
     ) -> Result<PublicKey, VaultFailError>;
     /// Remove a secret key from the vault
     fn secret_destroy(&mut self, context: Box<dyn Secret>) -> Result<(), VaultFailError>;
-    /// Compute Elliptic-Curve Diffie-Hellman using this secret key
-    /// and the specified uncompressed public key
-    fn ec_diffie_hellman(
-        &mut self,
-        context: &Box<dyn Secret>,
-        peer_public_key: &[u8],
-    ) -> Result<Box<dyn Secret>, VaultFailError>;
-    /// Compute Elliptic-Curve Diffie-Hellman using this secret key
-    /// and the specified uncompressed public key and return the HKDF-SHA256
-    /// output using the DH value as the HKDF ikm
-    fn ec_diffie_hellman_hkdf_sha256(
+}
+
+/// Trait with sign functionality
+pub trait SignerVault: Zeroize {
```

Comment:
Decided to have 2 traits, cause sometimes one party  may need only signing or verification (e.g. signing can be on server, and client should only verify signature)